src/routes/index.tsx

```diff
@@ -354,36 +354,50 @@ const RouterComp = ({ container, routeProps }: { container: CustomRouter.Route;
   return <container.comp {...routeProps} />
 }
 
-class PageErrorBoundary extends Component<
-  {},
-  {
-    error?: Error | null
-    info: {
-      componentStack?: string
-    }
+type PageErrorBoundaryState = {
+  error?: Error | null
+  info: {
+    componentStack?: string
   }
-> {
-  constructor(props: {}) {
+}
+
+type PageErrorBoundaryProps = {}
+
+class PageErrorBoundary extends Component<PageErrorBoundaryProps, PageErrorBoundaryState> {
+  constructor(props: PageErrorBoundaryProps) {
     super(props)
 
     this.state = {
-      error: undefined,
+      error: null,
       info: {
         componentStack: '',
       },
     }
   }
 
-  componentDidCatch(error: Error | null, info: object) {
+  static getDerivedStateFromError(error: Error) {
+    return { error }
+  }
+
+  componentDidUpdate(prevProps: PageErrorBoundaryProps) {
+    if (prevProps !== this.props) {
+      this.setState({ error: null, info: { componentStack: '' } })
```

Comment:
I think we need some comments to explain this piece of code and leave an issue address.